modules/virtualnetwork/variables.tf

```
@@ -173,6 +182,15 @@ DESCRIPTION
     error_message = "Address space entries must be specified in CIDR notation, e.g. 192.168.0.0/24."
   }
 
+  # validate ddos protection plan resource id for networks with ddos protection enabled
+  validation {
+    condition = alltrue([
+      for k, v in var.virtual_networks :
+      can(regex("^/subscriptions/[a-f\\d]{4}(?:[a-f\\d]{4}-){4}[a-f\\d]{12}/resourceGroups/[\\w-._]{1,89}[^\\s.]/providers/Microsoft.Network/ddosProtectionPlans/[\\w-_.]{2,64}$", v.ddos_protection_plan_id)) if v.ddos_protection_enabled
+    ])
+    error_message = "Hub network resource id must be an Azure virtual network resource id, e.g. /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/my-rg/providers/Microsoft.Network/ddosProtectionPlans/my-protection_plan."
```

Copy/paste error here. Should read:

```suggestion
    error_message = "Hub network resource id must be an Azure ddos protection plan resource id, e.g. /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/my-rg/providers/Microsoft.Network/ddosProtectionPlans/my-protection-plan."
```